custom_components/climate_group/climate.py

```diff
@@ -180,33 +181,78 @@ def should_poll(self) -> bool:
     
     async def async_update(self):
         """Query all members and determine the climate group state."""
-        all_states = [self.hass.states.get(x) for x in self._entity_ids]
-        states = list(filter(None, all_states))
-        self._target_temp = _reduce_attribute(states, 'temperature')
-        self._current_temp = _reduce_attribute(states, 'current_temperature')
-        self._min_temp = _reduce_attribute(states, 'min_temp', reduce=max)
-        self._max_temp = _reduce_attribute(states, 'max_temp', reduce=min)
-        self._mode = None
-        all_modes = [state.state for state in states]
-        if all_modes:
-            # Report the most common effect.
-            modes_count = Counter(itertools.chain(all_modes))
-            self._mode = modes_count.most_common(1)[0][0]
+        raw_states = [self.hass.states.get(x) for x in self._entity_ids]
+        all_states = list(filter(None, raw_states))
+
+        states = list(filter(lambda x: x is not None and 'preset_mode' in x.attributes, raw_states))
+
+        heat_states = list(filter(lambda x: x.state == HVAC_MODE_HEAT, states))
+        cool_states = list(filter(lambda x: x.state == HVAC_MODE_HEAT, states))
+
+        heat_actions = list(filter(lambda x: x.attributes['hvac_action'] == CURRENT_HVAC_HEAT, states))
+        cool_actions = list(filter(lambda x: x.attributes['hvac_action'] == CURRENT_HVAC_COOL, states))
+        idle_actions = list(filter(lambda x: x.attributes['hvac_action'] == CURRENT_HVAC_IDLE, states))
+
+        preset_away_stats = list(filter(lambda x: x.attributes['preset_mode'] == PRESET_AWAY, states))
+        preset_none_stats = list(filter(lambda x: x.attributes['preset_mode'] != PRESET_AWAY, states))
+        self._is_away = len(preset_away_stats) == len(states)
+
+        if self._is_away:
```

Comment:
Why is this split needed? Should the current/target temperature not be the average of all states?

Reply:
I found that if any thermostat was set to away, it skewed the averages. Effectively, in `away`, the thermostat is "off" unless the temperature gets very cold (or hot if using AC). My set up is 5 separate climate entities, connected to individual room's radiators. The `climate_group` then gives a quick and easy way to set the temperature for the whole house in one go. When I'm working away, the office is set to `away` as there is no point heating that room, and the temperature is set to 12 degrees. This showed the average target and actual temperatures as several degrees lower than they should be, if you disregarded this _inactive_ room. This was to try and compensate for that, using this logic:

- if all 'stats are `away`, use the average of all of them
- otherwise, use the average of all `not-away` stats

Reply:
Hmm, I think that your case is a somewhat special use-case. 
A `climate_group` is meant to control a group of thermostats as if it is one. So in theory all thermostats should be set to the exactly the same state in terms of control mode and target temperature. 
In your case, you should maybe have two separate groups, office and the rest. 

I do see that some people might want to exclude "off/away" thermostats, but if we introduce that behaviour, I would make it configurable.

Reply:
office was just an example. There are other rooms that may want to be disabled. :man_shrugging:

Reply:
Made this configurable:
```
    exclude:
    - away
    - boost
```